src/compat/responder.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const OS = require('os')
+const assert = require('assert')
+const MDNS = require('multicast-dns')
+const log = require('debug')('libp2p:mdns:compat:responder')
+const TCP = require('libp2p-tcp')
+const tcp = new TCP()
+const { SERVICE_TAG_LOCAL } = require('./constants')
+
+class Responder {
+  constructor (peerInfo) {
+    assert(peerInfo, 'missing peerInfo parameter')
+    this._peerInfo = peerInfo
+    this._peerIdStr = peerInfo.id.toB58String()
+    this._onQuery = this._onQuery.bind(this)
+  }
+
+  start (callback) {
+    this._mdns = MDNS()
+    this._mdns.on('query', this._onQuery)
+    setImmediate(() => callback())
+  }
+
+  _onQuery (event, info) {
+    const multiaddrs = tcp.filter(this._peerInfo.multiaddrs.toArray())
+    // Only announce TCP for now
+    if (!multiaddrs.length) return
+
+    const questions = event.questions || []
+
+    // Only respond to queires for our service tag
+    if (!questions.some(q => q.name === SERVICE_TAG_LOCAL)) return
+
+    log('got query', event, info)
+
+    const answers = []
+    const peerServiceTagLocal = `${this._peerIdStr}.${SERVICE_TAG_LOCAL}`
+
+    answers.push({
+      name: SERVICE_TAG_LOCAL,
+      type: 'PTR',
+      class: 'IN',
+      ttl: 120,
+      data: peerServiceTagLocal
+    })
+
+    // Only announce TCP multiaddrs for now
```

Comment:
Can you have as an action item of merging this PR to create an issue for tracking the announce of other transports?